Add Dialog model and related functionallity

TheLaw-tech opened the pull request:
ne pes patron

TheLaw-tech committed:
Add Dialog model and related functionallity

5x1fF1xgx1xti0n commented on Uanion/Uanion.Application/Features/Dialoge/Commands/CreateDialoge/CreateDialogeCommand.cs:
```diff
@@ -0,0 +1,8 @@
+
+public class CreateDialogeCommand : IRequest<Guid>
+{
+
```
Please, remove blank line here

5x1fF1xgx1xti0n commented on Uanion/Uanion.Application/Features/Dialoge/Commands/CreateDialoge/CreateDialogeCommandValidator.cs:
```diff
@@ -0,0 +1,28 @@
+{
+    public CreateDialogeCommandValidator()
+    {
+        RuleFor(p => p.Email)
```
This is a validator for CreateDialogCommand. There are no properties in CreateDialogCommand yet, therefore there should be no logic in validator for now

5x1fF1xgx1xti0n commented on Uanion/Uanion.Application/Features/Dialoge/Commands/CreateDialoge/CreateDialogeCommandHandler.cs:
```diff
@@ -0,0 +1,35 @@
+            throw new Exceptions.ValidationException(validatorResult);
+        }
+
+        var dialoge = _mapper.Map<Domain.Entities.Dialoge>(request);
```
That won't work. Please, register mapping profile for Dialog entity. Look Uanion.Application -> Profiles -> UserMappingProfile

5x1fF1xgx1xti0n commented on Uanion/Uanion.Application/Features/Dialoge/Queries/GetDialoge/DialogeViewModel.cs:
```diff
@@ -0,0 +1,7 @@
+public class DialogeViewModel
+{
+    public Guid DialogeId { get; set; }
+
```
Remove blank line here, pleae

5x1fF1xgx1xti0n commented on Uanion/Uanion.Application/Features/Dialoge/Queries/GetDialoge/GetDialogerQueryHandler.cs:
```diff
@@ -0,0 +1,25 @@
+    public async Task<DialogeViewModel> Handle(GetDialogeQuery request, CancellationToken cancellationToken)
+    {
+        var dialoge = await _dialogeRepository.GetByIdAsync(request.DialogeId);
+        var dialogeVm = _mapper.Map<DialogeViewModel>(dialoge);
```
That won't work. Please, register mapping profile for Dialog entity. Look Uanion.Application -> Profiles -> UserMappingProfile

5x1fF1xgx1xti0n reviewed (changes requested):
Please, rename "dialoge" to "dialog" everywhere.
Also some changes are missing:
1. As mentioned in comments below you have to add DialogMappingProfile
2. Add DialogController to API, ensure methods from the controller are displayed in SwaggerUI
3. Register DialogRepository to DI as it's done with UserRepository (look PersistenceServiceRegistration class)
4. Add Dialog DbSet to UanionDbContext
Thank you!

TheLaw-tech committed:
1.Renamed "dialoge" to "dialog" everywhere.
2.Added DialogMappingProfile.
3.Added DialogController to API.
4.Registered DialogRepository in DI.
5.Added Dialog DbSet to UanionDbContext.
6.Removed blank lines everywhere.
7.Fixed logic in CreateDialogCommand.

5x1fF1xgx1xti0n reviewed (approved):
Lgtm, thank you!

5x1fF1xgx1xti0n committed:
Merge branch 'dev' into dialoge-create-get
